chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/urluptime/DatadogSyntheticTestService.java

```diff
@@ -0,0 +1,146 @@
+package gov.healthit.chpl.scheduler.job.urluptime;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import com.datadog.api.client.ApiException;
+import com.datadog.api.client.v1.model.SyntheticsAPITest;
+import com.datadog.api.client.v1.model.SyntheticsAPITestConfig;
+import com.datadog.api.client.v1.model.SyntheticsAPITestType;
+import com.datadog.api.client.v1.model.SyntheticsAssertion;
+import com.datadog.api.client.v1.model.SyntheticsAssertionOperator;
+import com.datadog.api.client.v1.model.SyntheticsAssertionTarget;
+import com.datadog.api.client.v1.model.SyntheticsAssertionType;
+import com.datadog.api.client.v1.model.SyntheticsDeleteTestsPayload;
+import com.datadog.api.client.v1.model.SyntheticsTestDetails;
+import com.datadog.api.client.v1.model.SyntheticsTestOptions;
+import com.datadog.api.client.v1.model.SyntheticsTestOptionsHTTPVersion;
+import com.datadog.api.client.v1.model.SyntheticsTestOptionsScheduling;
+import com.datadog.api.client.v1.model.SyntheticsTestOptionsSchedulingTimeframe;
+import com.datadog.api.client.v1.model.SyntheticsTestRequest;
+
+import gov.healthit.chpl.util.DateUtil;
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2(topic = "serviceBasedUrlUptimeCreatorJobLogger")
+@Component
+public class DatadogSyntheticTestService {
+    private static final Integer HTTP_STATUS_OK = 200;
+    private static final String HTTP_METHOD_GET = "GET";
+    private static final Long SECONDS_IN_A_MINUTE = 60L;
+
+    private DatadogSyntheticTestApiProvider apiProvider;
+    private Boolean datadogIsReadOnly;
+    private String datadogTestStartTime;
+    private String datadogTestEndTime;
+    private Long datadogCheckEveryMinutes;
+    private Integer datadogTestTimeout;
+    private String datadogTestLocation;
+
+    public DatadogSyntheticTestService(DatadogSyntheticTestApiProvider apiProvider,
+            @Value("${datadog.syntheticTest.readOnly}") Boolean datadogIsReadOnly,
```

Comment:
Maybe another nitpicky naming thing.. I sometimes see "synthetics" plural and here in the properties it's "synthetic". I think I mostly hear us say the plural one when speaking so I'm wondering if that applies to the property names here, too.